src/python/dxpy/scripts/dx.py

```diff
@@ -3244,11 +3245,11 @@ def run_body(args, executable, dest_proj, dest_path, preset_inputs=None, input_n
     requested_nvidia_driver = cloned_nvidia_driver
 
     # Validate mutual exclusivity: instanceTypeSelector and clusterSpec cannot coexist
-    # Note: instanceType can override instanceTypeSelector, so we only check clusterSpec here
+    # Note: instanceType can override instanceTypeSelector, so we only raise error if no instanceType is present
     # instanceTypeSelector is build-time only and should never be in runtime systemRequirements
-    if cloned_instance_type_selector.as_dict() and requested_cluster_spec.as_dict():
-        raise err_exit("Cannot combine clusterSpec with instanceTypeSelector. "
-                      "instanceTypeSelector and clusterSpec are mutually exclusive.")
+    if cloned_instance_type_selector.as_dict() and requested_cluster_spec.as_dict() and not requested_instance_type.as_dict():
+        raise DXCLIError("Cannot combine clusterSpec with instanceTypeSelector without providing instanceType. "
+                        "instanceTypeSelector and clusterSpec are mutually exclusive.")
```

Comment:
it seems to be replicate the same check added above, both check 
```
if cloned_instance_type_selector.as_dict() and not requested_instance_type.as_dict():
```
and because only when the check above passes the requested_cluster_spec will be non null, no need to check all three again